stl/inc/xstring

```diff
@@ -1277,6 +1277,7 @@ public:
     template <_RANGES contiguous_range _Range>
         requires (_RANGES sized_range<_Range>
             && same_as<_RANGES range_value_t<_Range>, _Elem>
+            && !same_as<remove_cvref_t<_Range>, basic_string_view>

```

Comment:
I was under the mistaken impression that `basic_string_view` had defaulted move operations, which would avoid this problem: the non-template move constructor would be chosen over this constructor template. That said, I think it's a good idea for all converting constructors to be constrained not to accept arguments of the class's type to limit the potential for introducing constraint recursion under maintenance.

Could we please:
1. Add a defaulted move constructor and move assignment operator
2. Put the disqualification term first to minimize potential for constraint recursion:
```suggestion
    template <class _Range>
        requires (!same_as<remove_cvref_t<_Range>, basic_string_view>
            && _RANGES contiguous_range<_Range>
            && _RANGES sized_range<_Range>
            && same_as<_RANGES range_value_t<_Range>, _Elem>
```
and I'll submit an LWG issue with this as proposed resolution?

Reply:
After more thought while writing up the issue submission, I think we don't want to add the move operations after all. I suspect their omission is the simplest way to specify that moves leave the source object unmodified.